### PR TITLE
Pv avx512 sw

### PR DESCRIPTION
**Pull request by pnvaidya**
Updated SmithWaterman code to include the AVX512 optimizations.

### CODE REVIEW

File: src/main/java/com/intel/gkl/smithwaterman/IntelSmithWaterman.java [hunk 1]
```diff
@@ -92,6 +96,8 @@ public SWNativeAlignerResult align(byte[] refArray, byte[] altArray, SWParameter
 
         int offset = alignNative(refArray, altArray, cigar, parameters.getMatchValue(), parameters.getMismatchPenalty(), parameters.getGapOpenPenalty(), parameters.getGapExtendPenalty(), intStrategy);
 
+        //System.out.println(new String(cigar));
+
```
Comment (gspowley) [inline src/main/java/com/intel/gkl/smithwaterman/IntelSmithWaterman.java]: Please remove these lines or create a debug mode if they are useful.

File: src/main/native/smithwaterman/avx512-smithwaterman.h [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#include <stdint.h>
+
+#undef SIMD_ENGINE
+#define SIMD_ENGINE avx512
+
+#include "avx512-functions.h"
+#include "PairWiseSW.h"
```
Comment (gspowley) [inline src/main/native/smithwaterman/avx512-smithwaterman.h]: Add newline at end of file.

Comment (pnvaidya) [reply src/main/native/smithwaterman/avx512-smithwaterman.h]: Thanks. I have incorporated the changes you requested and updated the pull request below.

**Comment by erniebrau** [comment]
@pnvaidya @gspowley I added the checking for the flag (and tested it -- it works). I also added an INFO message which says when SW is using AVX512. This mimics what we do in PHMM, and it's helped a lot to be sure when AVX-512 is being used.

Thoughts?

**Comment by gspowley** [comment]
Good idea.

**Comment by pnvaidya** [comment]
Looks good. I will go ahead and merge it. And after testing will release it as 0.8.1

**Comment by pnvaidya** [comment]
AVX512 SmithWaterman addition and minor updates.

**Pull request merged**